src/equal.c

```diff
@@ -45,19 +49,33 @@ SEXP vctrs_equal_scalar(SEXP x, SEXP i_, SEXP y, SEXP j_, SEXP na_equal_) {
 }
 
 // [[ include("vctrs.h") ]]
-int equal_scalar_p(enum vctrs_type proxy_type,
-                   SEXP x, const void* x_p, R_len_t i,
-                   SEXP y, const void* y_p, R_len_t j,
-                   bool na_equal) {
+int equal_scalar_na_equal_p(enum vctrs_type proxy_type,
+                            SEXP x, const void* x_p, R_len_t i,
+                            SEXP y, const void* y_p, R_len_t j) {
+  switch (proxy_type) {
+  case vctrs_type_logical: return lgl_equal_scalar_na_equal(((const int*) x_p) + i, ((const int*) y_p) + j);
+  case vctrs_type_integer: return int_equal_scalar_na_equal(((const int*) x_p) + i, ((const int*) y_p) + j);
+  case vctrs_type_double: return dbl_equal_scalar_na_equal(((const double*) x_p) + i, ((const double*) y_p) + j);
+  case vctrs_type_complex: return cpl_equal_scalar_na_equal(((const Rcomplex*) x_p) + i, ((const Rcomplex*) y_p) + j);
+  case vctrs_type_character: return chr_equal_scalar_na_equal(((const SEXP*) x_p) + i, ((const SEXP*) y_p) + j);
+  case vctrs_type_raw: return raw_equal_scalar(((const Rbyte*) x_p) + i, ((const Rbyte*) y_p) + j, true);
```

Comment:
Yea maybe just for consistency could we have `raw_equal_scalar_na_equal()` that just does what `raw_equal_scalar()` does? Same with `raw_equal_scalar_na_propagate()`?